pages/base.py

```diff
@@ -15,89 +17,200 @@ def __init__(self, driver):
+
 
 class BasePage(BaseElement):
     url = None
 
-    def open(self):
+    def goto(self):
         self.driver.get(self.url)
+        self.check_page()
+
+    def check_page(self):
+        if not self.verify():
+            # handle any specific kind of error before go to page exception
+            self.error_handling()
+            raise PageException('Unexpected page structure: `{}` with identity {}'.format(
+                self.driver.current_url, self.locators['identity']
+            ))
+
+    def verify(self):
+        try:
+            self.identity
+        except ValueError:
+            return False
+        else:
+            return True
+
+    def error_handling(self):
+        pass
+
+    def reload(self):
+        self.driver.refresh()
+
+
+class Navbar(BaseElement):
+
+    locators = {
+        'service_dropdown': (By.CSS_SELECTOR, '#navbarScope > div.container > div.navbar-header > div.dropdown > button'),
+        'home_link': (By.CSS_SELECTOR, '#navbarScope > div.container > div.navbar-header > div.dropdown > ul > li:nth-child(1) > a'),
+        'preprints_link': (By.CSS_SELECTOR, '#navbarScope > div.container > div.navbar-header > div.dropdown > ul > li:nth-child(2) > a'),
+        'registries_link': (By.CSS_SELECTOR, '#navbarScope > div.container > div.navbar-header > div.dropdown > ul > li:nth-child(3) > a'),
+        'meetings_link': (By.CSS_SELECTOR, '#navbarScope > div.container > div.navbar-header > div.dropdown > ul > li:nth-child(4) > a'),
+        'search_link':(By.ID, 'navbar-search'),
+        'support_link': (By.ID, 'navbar-support'),
+        'donate_link': (By.ID, 'navbar-donate'),
+        'user_dropdown': (By.CSS_SELECTOR, '#secondary-navigation > ul > li:nth-last-of-type(1) > button'),
+        'user_dropdown_profile': (By.CSS_SELECTOR, '#secondary-navigation > ul > li.dropdown.open > ul > li:nth-child(1) > a'),
+        'user_dropdown_support': (By.CSS_SELECTOR, '#secondary-navigation > ul > li.dropdown.open > ul > li:nth-child(2) > a'),
+        'user_dropdown_settings': (By.CSS_SELECTOR, '#secondary-navigation > ul > li.dropdown.open > ul > li:nth-child(3) > a'),
+        'sign_up_button': (By.LINK_TEXT, 'Sign Up'),
+        'sign_in_button': (By.LINK_TEXT, 'Sign In'),
+        'logout_link': (By.CSS_SELECTOR, '#secondary-navigation > ul > li.dropdown.open > ul > li:nth-child(4) > a'),
+        'current_service': (By.CSS_SELECTOR, '#navbarScope > div > div > div.service-home > a > span.current-service > strong')
+    }
+
+    def verify(self):
+        return len(self.find_elements(By.ID, 'navbarScope')) == 1
+
+    def is_logged_in(self):
+        try:
+            if self.sign_up_button:
+                return False
+        except ValueError:
+            return True
+
+
+class LoginPage(BasePage):
+    url = settings.OSF_HOME + '/login'
+
+    locators = {
+        'identity': (By.XPATH, '/html/body[@id="cas"]/div[@id="container"]', settings.LONG_TIMEOUT),
```

Comment:
See if you can get Longze to put in an ID if this is on CAS. Might also need to do fakecas for local testing?

Reply:
Fakecas use a different identity, which is configured here. https://github.com/CenterForOpenScience/OSF-Integration-Tests/pull/14/files#diff-794046ac4b020df503b67eb67973abe0R134